userspace/libscap/engine/gvisor/gvisor.cpp

```diff
@@ -151,27 +151,27 @@ static int32_t gvisor_getpid_global(struct scap_engine_handle engine, int64_t* p
 #endif
 
 extern const struct scap_vtable scap_gvisor_engine = {
-	"gvisor",
-	SCAP_MODE_LIVE,
-	NULL,
-
-	gvisor_match,
-	gvisor_alloc_handle,
-	gvisor_init,
-	gvisor_free_handle,
-	gvisor_close,
-	gvisor_next,
-	gvisor_start_capture,
-	gvisor_stop_capture,
-	gvisor_configure,
-	gvisor_get_stats,
-	gvisor_get_n_tracepoint_hit,
-	gvisor_get_n_devs,
-	gvisor_get_max_buf_used,
-	gvisor_get_threadlist,
-	gvisor_get_threadinfos,
-	gvisor_get_fdinfos,
-	gvisor_get_vxid,
-	gvisor_get_vxid,
-	gvisor_getpid_global
+	.name = "gvisor",
```

Comment:
https://github.com/falcosecurity/libs/pull/519
Aren't we re-introducing the issue?

Reply:
The issue is only with incomplete definitions (if you initialize only a subset of the fields). The problematic initialization in https://github.com/falcosecurity/libs/pull/511 was only populating a subset of the fields.

Using C99 style for the vtable is actually valuable, because it breaks the build on incomplete vtable initializers and I wish we could have the same in the C-based engines.

TBH I should check how our ancient gcc 4.8 (undoubtedly the oldest compiler still used to build the libs) handles this. I'll run a test soon(ish) and report back.

Reply:
Any news? :)

Reply:
In a surprising plot twist, I couldn't get a sample to fail, except with `-pedantic`:

```
[root@658f5af52bf5 ~]# g++ struct.cc -std=c++0x -o struct -Wall -O2 -Werror -pedantic
struct.cc: In function ‘int main()’:
struct.cc:10:14: error: ISO C++ does not allow C99 designated initializers [-Werror=pedantic]
  foo foo_x = {
              ^
struct.cc:11:9: error: ISO C++ does not allow C99 designated initializers [-Werror=pedantic]
   .a = 1,
         ^
cc1plus: all warnings being treated as errors
[root@658f5af52bf5 ~]# g++ struct.cc -std=c++0x -o struct -Wall -O2 -Werror
[root@658f5af52bf5 ~]# cat struct.cc
#include <cstdio>

struct foo {
	int a;
	int b;
	int* c;
};

int main() {
	foo foo_x = {
		.a = 1,
		.b = 2,
	};

	printf("x.a=%d\n", foo_x.a);
}
```

(same if I wrap the struct definition in an `extern "C" {}` block)

🤷

Reply:
Then this LGTM!